bridge/client/app/_models/trace.ts

```diff
@@ -1,32 +1,72 @@
+  TESTS_FINISHED = 'sh.keptn.events.tests-finished',
+  START_EVALUATION = 'sh.keptn.event.start-evaluation',
+  EVALUATION_DONE = 'sh.keptn.events.evaluation-done',
+  START_SLI_RETRIEVAL = 'sh.keptn.internal.event.get-sli',
+  SLI_RETRIEVAL_DONE = 'sh.keptn.internal.event.get-sli.done',
+  DONE = 'sh.keptn.events.done',
+  PROBLEM_OPEN = 'sh.keptn.event.problem.open',
+  PROBLEM_DETECTED = 'sh.keptn.events.problem',
+  PROBLEM_RESOLVED = 'sh.keptn.events.problem.resolved',
+  PROBLEM_CLOSED = 'sh.keptn.event.problem.close',
+  APPROVAL_TRIGGERED = 'sh.keptn.event.approval.triggered',
+  APPROVAL_FINISHED = 'sh.keptn.event.approval.finished'
 };
-let icons = {
-  "sh.keptn.event.configuration.change": "duplicate",
-  "sh.keptn.events.deployment-finished": "deploy",
-  "sh.keptn.events.tests-finished": "perfromance-health",
-  "sh.keptn.event.start-evaluation": "traffic-light",
-  "sh.keptn.events.evaluation-done": "traffic-light",
-  "sh.keptn.internal.event.get-sli": "collector",
-  "sh.keptn.internal.event.get-sli.done": "collector",
-  "sh.keptn.event.problem.open": "criticalevent",
-  "sh.keptn.events.problem": "criticalevent",
-  "sh.keptn.event.problem.close": "applicationhealth"
+
+enum ResultTypes {
+  PASSED = 'pass',
+  WARNING = 'warning',
+  FAILED = 'fail'
+};
+
+enum ProblemStates {
+  RESOLVED = 'RESOLVED'
 };
 
+enum ApprovalStates {
+  APPROVED = 'pass',
+  DECLINED = 'failed'
+};
+
+const EVENT_LABELS = {
+  [EventTypes.SERVICE_CREATE]: "Service create",
+  [EventTypes.CONFIGURATION_CHANGE]: "Configuration change",
+  [EventTypes.CONFIGURE_MONITORING]: "Configure monitoring",
+  [EventTypes.DEPLOYMENT_FINISHED]: "Deployment finished",
+  [EventTypes.TESTS_FINISHED]: "Tests finished",
+  [EventTypes.START_EVALUATION]: "Start evaluation",
+  [EventTypes.EVALUATION_DONE]: "Evaluation done",
+  [EventTypes.START_SLI_RETRIEVAL]: "Start SLI retrieval",
+  [EventTypes.SLI_RETRIEVAL_DONE]: "SLI retrieval done",
+  [EventTypes.DONE]: "Done",
+  [EventTypes.PROBLEM_OPEN]: "Problem open",
+  [EventTypes.PROBLEM_DETECTED]: "Problem detected",
+  [EventTypes.PROBLEM_RESOLVED]: "Problem resolved",
+  [EventTypes.PROBLEM_CLOSED]: "Problem closed",
+  [EventTypes.APPROVAL_TRIGGERED]: "Approval triggered",
+  [EventTypes.APPROVAL_FINISHED]: "Approval finished"
+};
+const EVENT_ICONS = {
+  [EventTypes.CONFIGURATION_CHANGE]: "duplicate",
+  [EventTypes.DEPLOYMENT_FINISHED]: "deploy",
+  [EventTypes.TESTS_FINISHED]: "perfromance-health",
+  [EventTypes.START_EVALUATION]: "traffic-light",
+  [EventTypes.EVALUATION_DONE]: "traffic-light",
+  [EventTypes.START_SLI_RETRIEVAL]: "collector",
+  [EventTypes.SLI_RETRIEVAL_DONE]: "collector",
+  [EventTypes.PROBLEM_OPEN]: "criticalevent",
+  [EventTypes.PROBLEM_DETECTED]: "criticalevent",
+  [EventTypes.PROBLEM_CLOSED]: "applicationhealth",
+  [EventTypes.APPROVAL_TRIGGERED]: "unknown",
+  [EventTypes.APPROVAL_FINISHED]: "checkmark"
+};
+
+const DEFAULT_ICON = "information";
```

Comment:
if this enums and constants located here are used in other classes then they should be extracted in a different .ts file where all the constants are reunited.

Reply:
Yep, I separated the enums and constants in different files.